README.md

```diff
@@ -1,2 +1,193 @@
+
+## Setup guide
+
+To use the Asana Connector in Ballerina, you must have an Asana account and a Personal Access Token (PAT) or OAuth2 credentials for authentication.
+
+If you already have an Asana account, you can integrate the connector with your existing account. If not, you can create a new Asana account by visiting [Asana's Sign Up page](https://asana.com/) and following the registration process.
+Once you have an Asana account, you can proceed to create a PAT or set up OAuth2.
+
+### Step 1: Access Asana developer console
+
+1. Log in to your Asana account.
+2. After logging in, navigate to the [Asana developer console](https://app.asana.com/0/my-apps).
+3. Click on the **+ Create new token** button.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/1-developer-console.png alt="Asana Developer Console">
+
+### Step 2: Create a new access token
+
+1. Provide a name for the token and accept Asana's API terms checkbox after reading them.
+2. Click on the **Create token** button.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/2-generate-token.png alt="Generate new PAT">
+
+### Step 5: Generate the token
+
+1. Copy the generated token and keep it secure. You will need this token to authenticate the Asana connector.
+
+   <img src=https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-asana/master/docs/setup/resources/3-copy-token.png alt="Copy PAT">
+
+## Quickstart
+
+To use the `Asana` connector in your Ballerina application, modify the `.bal` file as follows:
+
+### Step 1: Import the connector
+
+Import the `ballerinax/asana` package into your Ballerina project.
+
+```ballerina
+import ballerinax/asana;
+```
+
+### Step 2: Instantiate a new connector
+
+Create an `asana:ConnectionConfig` with the obtained PAT (or OAuth2) credentials and initialize the connector with it.
+
+```ballerina
+asana:ConnectionConfig asanaConfig = {
+    auth: {
+        token: authToken
+    }
+};
+
+asana:Client asana = check new (asanaConfig);
+```
+
+### Step 3: Invoke the connector operation
+
+Now, utilize the available connector operations.
+
+#### Get all projects for the authenticated user
+
+```ballerina
+Inline_response_200_19 asanaProjects = check asana->/projects();
+```
+
+#### Create a new task in a project
+
+```ballerina
+asana:Tasks_body taskReq = {
+    data: {
+        name: "Email Marketing Campaign",
+        notes: "Create a new email marketing campaign for the upcoming product launch.",
+        workspace: "<workspaceId>",
+        projects: ["<projectId>"]
+    }
+};
+
+var createdTask = check asana->/tasks.post(taskReq);
```

Comment:
Can't we use the expected type rather than using `var` ?

Reply:
This was added intentionally for the sake of simplicity, as the return type of the particular API is `asana:Inline_response_201_7` (as most of the operations defined in the Asana's official OpenAPI contracts are having inline responses). Already had a chat on this with @lnash94 as well, but it seems we don't have an alternative for that.

Reply:
Agreed on your point. But if we think about the usability of this API, developers have to use this inline type to do something useful with the response. 

What if we do something like this:
```ballerina
record { TaskResponse data?; } taskCreated = check asana->/tasks.post(taskReq);
```
I am not fully sure whether we are adding extra complexity with above. Just thinking out loud.

Reply:
Your suggested approach looks much better compared to using the auto-generated name or `var`. And this representation will clearly conveys the idea that the real data is encapsulated within the response under the `data` field.